### ISSUE TITLE
Trouble installing

### ISSUE THREAD

**Issue by spinningsand** [Davincible/gotiktoklive#3]
When I try to install, it outputs:

_go: go.mod file not found in current directory or any parent directory.
        'go get' is no longer supported outside a module.
        To build and install a command, use 'go install' with a version,
        like 'go install example.com/cmd@latest'
        For more information, see https://golang.org/doc/go-get-install-deprecation
        or run 'go help get' or 'go help install'._

So then I use `go install github.com/Davincible/gotiktoklive@latest` and get:

_go: downloading github.com/Davincible/gotiktoklive v0.0.0-20220523132158-127857a3f5b6
go: github.com/Davincible/gotiktoklive@latest (in github.com/Davincible/gotiktoklive@v0.0.0-20220523132158-127857a3f5b6):
        The go.mod file for the module providing named packages contains one or
        more replace directives. It must not contain directives that would cause
        it to be interpreted differently than if it were the main module._

**Comment by Davincible** [comment]
`go install` is typically used to install commands, as this package doesn't provide commands (yet), you can install using `go get`.

Make sure to be inside your module directory;
`go mod init mytiktok`
 
And then install `go get ....@master`

**Comment by spinningsand** [comment]
Thanks, it worked. But how do I use it? I think you have to go through the .go files and edit them to fit what you want and then run it in the command line, but I'm not for certain.

**Comment by spinningsand** [comment]
Am I on the right track? Sorry, I'm new to this, so I'm a little lost. I tried running a package, but it said:

_package command-line-arguments is not a main package_

So I tried to find the main package, but can't and I'm not for certain if I'm approaching it right to begin with.

**Comment by Davincible** [comment]
What is your goal? Just to download streams

**Comment by spinningsand** [comment]
Yes, I was wanting to download streams that I couldn't be there to watch when they're live.

**Comment by Davincible** [comment]
create a `main.go` file with the following, and modify to your needs;

```go
package main

import "github.com/Davincible/gotiktoklive"

func main() {
    <READE.md example code>
}
```

Then `go run main.go`

If you want it to not print out all the events happening, remove all the print statements.

**Comment by spinningsand** [comment]
Thanks. This includes proxies so it doesn't block your IP, right?

**Comment by Davincible** [comment]
No, by default it uses your own IP. You can set a proxy manually, as described in the README.md.

Although you'd have to do something crazy to get blocked

**Comment by spinningsand** [comment]
I created a `main.go` file within the tiktoklive folder. First I did `cd C:\Users\CLUNKYBOXCOMPUTER\go\pkg\mod\github.com\!davincible\gotiktoklive@v0.0.0-20220525135242-dda499f27e6d`  and then I ran `go run main.go` and got:

_CreateFile main.go: The system cannot find the file specified._

Are the contents on `main.go` correct? I didn't alter anything because I was wanting to see if it would work. Here:

```
package main

import "github.com/Davincible/gotiktoklive"

func main() {
    // Create TikTok Instance
tiktok := gotiktoklive.NewTikTok()

// Track a TikTok user by username
live, err := tiktok.TrackUser("promobot.robots")
if err != nil {
    panic(err)
}

// Start downloading stream
// Make sure you have the ffmpeg binary installed, and present in your path.
if err := live.DownloadStream(); err != nil {
    panic(err)
}

// Receive livestream events through the live.Events channel
for event := range live.Events {
    switch e := event.(type) {

    // You can specify what to do for specific events. All events are listed below.
    case gotiktoklive.UserEvent:
        fmt.Printf("%T : %s %s\n", e, e.Event, e.User.Username)

    // List viewer count
    case gotiktoklive.ViewersEvent:
        fmt.Printf("%T : %d\n", e, e.Viewers)

    // Specify the action for all remaining events
    default:
        fmt.Printf("%T : %+v\n", e, e)
    }
}

// TikTok allows you to track and discover current live streams.
type TikTok struct {

	Debug bool

	// LogRequests when set to true will log all made requests in JSON to debugHandler
	LogRequests bool
}

// NewTikTok creates a tiktok instance that allows you to track live streams
//  and discover current livestreams.
func NewTikTok() *TikTok {}

// TrackUser will start to track the livestream of a user, if live.
// To listen to events emitted by the livestream, such as comments and viewer
//  count, listen to the Live.Events channel.
func (t *TikTok) TrackUser(username string) (*Live, error) {}

// GetUserInfo will fetch information about the user, such as follwers stats,
//  their user ID, as well as the RoomID, with which you can tell if they are live.
func (t *TikTok) GetUserInfo(user string) (*UserInfo, error) {}

// GetPriceList fetches the price list of tiktok coins. Prices will be given in
//  USD cents and the cents equivalent of the local currency of the IP location.
// To fetch a different currency, use a VPN or proxy to change your IP to a
//  different country.
func (t *TikTok) GetPriceList() (*PriceList, error) {}

// NewFeed creates a new Feed instance. Start fetching reccomended livestreams
//  with Feed.Next().
func (t *TikTok) NewFeed() *Feed {}

func (t *TikTok) SetDebugHandler(f func(...interface{})) {}

func (t *TikTok) SetErrorHandler(f func(error)) {}

func (t *TikTok) SetInfoHandler(f func(...interface{})) {}

func (t *TikTok) SetWarnHandler(f func(...interface{})) {}

// SetProxy will set a proxy for both the http client as well as the websocket.
// You can manually set a proxy with this method, or by using the HTTPS_PROXY
//  environment variable.
// ALL_PROXY can be used to set a proxy only for the websocket.
func (t *TikTok) SetProxy(url string, insecure bool) error {}
}

```

I didn't alter anything because I was wanting to see if it would work at all.

**Comment by Davincible** [comment]
Hahaha not exactly right. You're not supposed to copy the _entire_ `README.md`, just the example in the beginning. 

I recommend you watch some Go programming tutorials on YouTube to get going with the basics. 

Otherwise, I'm available for hire to create a solution that fits your needs or to teach you the way.

I'm afraid I can't fully guide you in a GitHub issue, so I will close this issue.

**Comment by spinningsand** [comment]
I did originally copy the example in the beginning but I got the same output, so that's when I included the second part. I'm still not sure how to run the proxy even when I do get it going, but this has gone way beyond the original issue anyway. Thanks for helping me out. This stuff can be confusing when you don't know anything about programming. I can run some commands in cmd and that's about it lol. I'll try and learn some Go programming and get back to this and if I still don't understand it I might hire you to help me out lol. Thanks again.
